blackjax/mcmc/trajectory.py

```diff
@@ -36,7 +36,9 @@
 memory by keeping states that will subsequently be discarded.
 
 """
-from typing import Callable, NamedTuple, Tuple
+from __future__ import annotations
```

Comment:
Not sure what is the use for replacing typing's `Tuple` with `tuple` and using `annotations`, but it seems like a change that should be done separately to this PR.